### ISSUE TITLE
There was an error when attempting to execute the process '/usr/bin/git'.

### ISSUE THREAD

**Issue by mattBrzezinski** [peter-evans/create-pull-request#534]
### Subject of the issue
Stacktrace from my GH Action:

```
Successfully installed GitPython-3.1.2 PyGithub-1.51 certifi-2020.6.20 chardet-3.0.4 deprecated-1.2.10 gitdb-4.0.5 idna-2.9 pyjwt-1.7.1 requests-2.24.0 setuptools-46.4.0 smmap-3.0.4 urllib3-1.25.9 wheel-0.34.2 wrapt-1.12.1
/usr/bin/git config --local --name-only --get-regexp http.https://github.com/.extraheader ^AUTHORIZATION:
##[error]There was an error when attempting to execute the process '/usr/bin/git'. This may indicate the process failed to start. Error: spawn /usr/bin/git ENOENT
```

### Steps to reproduce
[GitHub Action](https://github.com/mattBrzezinski/JuliaCloud-AWS-State/blob/a488960d67471c4e9cabc5b8664563774739e578/.github/workflows/update_apis.yml)

I have another project (JuliaCloud/AWS.jl)[https://github.com/JuliaCloud/AWS.jl], instead of spamming merge requests there I wanted to use a different repo to test running some code from it.

**Comment by mattBrzezinski** [comment]
I have attempted to use `v3` and it results in a similar error.

```
 Create Pull Request0s
    draft: false
Run peter-evans/create-pull-request@v3
  with:
    path: ~/.julia/dev/AWS/
    token: ***
    commit-message: [create-pull-request] automated change
    committer: GitHub <noreply@github.com>
    author: mattBrzezinski <mattBrzezinski@users.noreply.github.com>
    signoff: false
    branch: create-pull-request/patch
    title: Changes by create-pull-request action
    body: Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action
    draft: false
Save persisted git credentials
  /usr/bin/git config --local --name-only --get-regexp http.https://github.com/.extraheader ^AUTHORIZATION:
  ##[error]There was an error when attempting to execute the process '/usr/bin/git'. This may indicate the process failed to start. Error: spawn /usr/bin/git ENOENT
Restore persisted git credentials
  /usr/bin/git config --local --name-only --get-regexp http.https://github.com/.extraheader ^AUTHORIZATION:
  ##[error]There was an error when attempting to execute the process '/usr/bin/git'. This may indicate the process failed to start. Error: spawn /usr/bin/git ENOENT
```

**Comment by peter-evans** [comment]
Hi @mattBrzezinski 

The issue is that you need to checkout the repository that will be the target of the pull request. The action needs a working repository to be checked out and expects that to be done with the official `actions/checkout@v2` action.

```yml
    steps:
      - uses: actions/checkout@v2
        with:
          repository: JuliaCloud/AWS.jl

      - uses: julia-actions/setup-julia@latest
        with:
          version: ${{ matrix.julia-version }}

      - name: UpdateAPI
        # Run against the latest version, since everything merged into master will be tagged.
        run: sudo julia -e 'import Pkg; Pkg.add(Pkg.PackageSpec(url="https://github.com/JuliaCloud/AWS.jl.git", rev="master")); using AWS; AWS.AWSMetadata.parse_aws_metadata();'

      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3
        with:
         token: ${{ secrets.PAT }}
```
I'm not sure what the `UpdateAPI` step is doing. It probably needs modifying to update files in the already checked out repository.

You are creating a PR in a remote repository (not the repository were the workflow is running) and so you need to use a PAT. See the [documentation here](https://github.com/peter-evans/create-pull-request/blob/master/docs/concepts-guidelines.md#creating-pull-requests-in-a-remote-repository).

Note that if you are planning to expand the strategy matrix to run multiple jobs, you need to make sure that each job creates a unique PR branch. You can do that by using the matrix variables in the branch name.
```yml
      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3
        with:
         token: ${{ secrets.PAT }}
         branch: update-api-${{ matrix.os }}-${{ matrix.julia-version }}-${{ matrix.julia-arch }}
```

**Comment by mattBrzezinski** [comment]
Hi @peter-evans 

This is wonderful, I just started looking into the documentation for remote repository and realized that was a mistake. I was originally using `create-pull-request@v2` and had a working example, although that was one revision and 6 months ago.

Thank you for clarifying all of this information! I'm going to leave the issue open for the next little bit in case I have additional questions. But I'll close it off afterwards.

**Comment by mattBrzezinski** [comment]
Thanks again for the feedback, definitely helped unblock our project. 😄 

Resolving this issue now!
